### PR TITLE
First task from testing with mocks

### PR DESCRIPTION
**Pull request by p-nedelchev**
Needs a review

### CODE REVIEW

File: tests/EndpointFilterTest.java [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+        context.checking(new Expectations() {{
+            oneOf(endpoint).matches(url);
+            will(returnValue(true));
+        }});
+        assertThat(filter.shouldFilter(url), is(true));
+    }
+
+    @Test
+    public void startingWithKeyword () {
+        StartsWithKeyword keyword = new StartsWithKeyword("www");
```
Comment (EmilGeorgiev) [inline tests/EndpointFilterTest.java]: this should be tested in another test class

File: src/taskone/EndpointFilter.java [hunk 2]
```diff
@@ -0,0 +1,21 @@
...
+       this.endpoints = endpoints;
+    }
+
+    public boolean shouldFilter (String url) {
+        for (Endpoint each : endpoints) {
+            if (each.matches(url)) {
+                return true;
+            }
+        }
+        return false;
```
Comment (EmilGeorgiev) [inline src/taskone/EndpointFilter.java]: this is not tested

File: tests/EndpointFilterTest.java (hunk 1 above)
Comment (p-nedelchev) [inline tests/EndpointFilterTest.java]: Done. I tested it in another class.

File: src/taskone/EndpointFilter.java (hunk 2 above)
Comment (p-nedelchev) [inline src/taskone/EndpointFilter.java]: Done. I tested false case too.

**Comment by EmilGeorgiev** [comment]
LGTM

**Pull request merged**